### ISSUE TITLE
bug(expr): `to_char` panic with timestamp and null

### ISSUE THREAD

**Issue by kwannoel** [risingwavelabs/risingwave#7327]
### Describe the bug

```
thread 'risingwave-main' panicked at 'called `Result::unwrap()` on an `Err` value: ProstFieldNotFound("rex_node")', src/expr/src/expr/build_expr_from_prost.rs:237:69
...
   4: risingwave_expr::expr::build_expr_from_prost::build_to_char_expr
             at ./src/expr/src/expr/build_expr_from_prost.rs:237:44
   5: risingwave_expr::expr::build_from_prost
             at ./src/expr/src/expr/mod.rs:121:19
```

View full logs below `frontend.log`.

### To Reproduce

```sql
select to_char(TIMESTAMP '2022-12-02 01:30:34', CAST(NULL AS CHARACTER VARYING));
```

### Additional context

[compute-node-5688.log](https://github.com/risingwavelabs/risingwave/files/10394742/compute-node-5688.log)
[frontend-4566.log](https://github.com/risingwavelabs/risingwave/files/10394743/frontend-4566.log)
[meta-node-5690.log](https://github.com/risingwavelabs/risingwave/files/10394744/meta-node-5690.log)
[risedev.log](https://github.com/risingwavelabs/risingwave/files/10394745/risedev.log)

**Comment by kwannoel** [comment]
Currently worked around in sqlsmith

**Comment by xxchan** [comment]
fix by #7936?!
